src/exm-unified-data.c

```diff
@@ -0,0 +1,315 @@
+                                   ExmExtension     *extension)
+{
+    self->local_data = g_object_ref (extension);
+}
+
+ExmSearchResult *
+exm_unified_data_get_web_data (ExmUnifiedData *self)
+{
+    return self->web_data;
+}
+
+void
+exm_unified_data_set_web_data (ExmUnifiedData *self,
+                                 ExmSearchResult  *extension)
+{
+    self->web_data = g_object_ref (extension);
+}
+
+const char *
+exm_unified_data_get_name (ExmUnifiedData *self)
+{
+    const char *name;
+
+    if (self->web_data)
+    {
+        g_object_get (self->web_data, "name", &name, NULL);
+        return name;
+    }
+
+    if (self->local_data)
+    {
+        g_object_get (self->local_data, "display-name", &name, NULL);
+        return name;
+    }
+
+    return NULL;
+}
+
+const char *
+exm_unified_data_get_creator (ExmUnifiedData *self)
+{
+    const char *creator;
+
+    if (self->web_data)
+    {
+        g_object_get (self->web_data, "creator", &creator, NULL);
+        return creator;
+    }
+
+    return NULL;
+}
+
+const char *
+exm_unified_data_get_uuid (ExmUnifiedData *self)
+{
+    const char *uuid;
+
+    if (self->web_data)
+    {
+        g_object_get (self->web_data, "uuid", &uuid, NULL);
+        return uuid;
+    }
+
+    if (self->local_data)
+    {
+        g_object_get (self->local_data, "uuid", &uuid, NULL);
+        return uuid;
+    }
+
+    return NULL;
+}
+
+const char *
+exm_unified_data_get_description (ExmUnifiedData *self)
+{
+    const char *description;
+
+    if (self->web_data)
+    {
+        g_object_get (self->web_data, "description", &description, NULL);
+        return description;
+    }
+
+    if (self->local_data)
+    {
+        g_object_get (self->local_data, "description", &description, NULL);
+        return description;
+    }
+
+    return NULL;
+}
+
+gboolean
+exm_unified_data_get_screenshot_uri (ExmUnifiedData  *self,
+                                     char           **uri)
+{
+    g_return_val_if_fail (uri != NULL, FALSE);
+
+    *uri = NULL;
+
+    if (self->web_data)
+    {
+        const char *tmp;
+
+        g_object_get (self->web_data, "screenshot", &tmp, NULL);
+        *uri = g_strdup (tmp);
+
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+gboolean
+exm_unified_data_get_link (ExmUnifiedData  *self,
+                           char           **link)
+{
+    g_return_val_if_fail (link != NULL, FALSE);
+
+    *link = NULL;
+
+    if (self->web_data)
+    {
+        const char *tmp;
+
+        g_object_get (self->web_data, "link", &tmp, NULL);
+        *link = g_strdup (tmp);
+
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+gboolean
+exm_unified_data_get_homepage (ExmUnifiedData  *self,
+                               char           **homepage)
+{
+    g_return_val_if_fail (homepage != NULL, FALSE);
+
+    *homepage = NULL;
+
+    if (self->web_data)
+    {
+        const char *tmp;
+
+        g_object_get (self->web_data, "homepage", &tmp, NULL);
```

Comment:
```suggestion
        g_object_get (self->web_data, "url", &tmp, NULL);
```

This fixes:

> * [ ] Fix homepage link (NOT IMPLEMENTED)

Maybe I should have used `url`, but since there is `link` already, using `homepage` seemed better to distinguish them.